src/Microsoft.OData.Edm/Validation/ValidationRules.cs

```diff
@@ -924,28 +924,62 @@ public static class ValidationRules
                     }
                 });
 
+        // Remove this in 8.x. It is superseded by EntityTypeInvalidKeyKeyDefinedInAncestor
         /// <summary>
         /// Validates that a key is not defined if there is already a key in the base type.
         /// </summary>
+        [Obsolete("Please use EntityTypeInvalidKeyKeyDefinedInAncestor")]
         public static readonly ValidationRule<IEdmEntityType> EntityTypeInvalidKeyKeyDefinedInBaseClass =
+            new ValidationRule<IEdmEntityType>(
+               (context, entityType) =>
+               {
+                   if (entityType.BaseType != null &&
+                       entityType.DeclaredKey != null &&
+                       entityType.BaseType.TypeKind == EdmTypeKind.Entity &&
+                       entityType.BaseEntityType().DeclaredKey != null)
+                   {
+                       context.AddError(
+                       entityType.Location(),
+                       EdmErrorCode.InvalidKey,
+                       Strings.EdmModel_Validator_Semantic_InvalidKeyKeyDefinedInBaseClass(entityType.Name, entityType.BaseEntityType().Name));
+                   }
+               });
+
+        /// <summary>
+        /// Validates that a key is not more than once in a type hierarchy
+        /// </summary>
+        public static readonly ValidationRule<IEdmEntityType> EntityTypeInvalidKeyKeyDefinedInAncestor =

```

Comment:
Why double "key" in the naming?

Reply:
I followed the convention of the previous test. I believe it's intended to be parsed with a space between the two keys, as in: "InvalidKey, KeyDefinedInAncestor"